### PR TITLE
dedupe tag_ids on task model

### PR DESCRIPTION
**Pull request by nlandolfi**
Fixes issue mentioned [here](https://github.com/viggyfresh/LifeDB/pull/1#issuecomment-290642606).

From a comment left to myself in the codebase:
```
// TODO(nclandolfi): this is easy to forget,
// (adding the dedupe). Is there a better solution?
// Generalizing record clean up? Why does protobuf merge
// the arrays? Is that desired behavior? Merging two sets?
// How to declare repeated set?
```

A (incomplete) list of alternative solutions:
 * Use maps from string to bool, explicitly indicating set types

### CODE REVIEW

**Comment by nlandolfi** [comment]
Curious for you opinion on this @viggyfresh

File: models/integrity.go [hunk 1]
```diff
@@ -558,3 +566,22 @@ func (b *Bounty) Dedupe() {
...
 }
+
+func (t *Task) Dedupe() {
+	if t == nil {
+		return
+	}
+
+	if len(t.TagIds) > 0 {
+		j := 0
+		seen := map[string]bool{}
```
Comment (viggyfresh) [inline models/integrity.go]: In lieu of a provided set class, this is I think every language's built in approximation of sets. Seems reasonable

Comment (nlandolfi) [inline models/integrity.go]: a list is or a map is?

Comment (viggyfresh) [inline models/integrity.go]: maps!

Comment (nlandolfi) [inline models/integrity.go]: should I convert the protobuf type to be a map for things that are explicitly sets? Or is the behavior always that when you set an array it should replace the current array, not append? I feel like if you can only ever append then you could never reset. With maps I could see setting keys to zero, but then I would probably add a cleaner on those that removed all false keys which would be similar to deduplication here. Perhaps it should be a clean override for arrays when we merge the data? Would there ever be the case of a gian array that you didn't want to have the specify the whole thing but just append to it?

Comment (viggyfresh) [inline models/integrity.go]: I don't ever see a case for having a truly giant array...at least not as far as I can tell. Someone who says "tag_ids": ["1"], I would think, expects `tag_ids` to be set to `["1"]`. I think replacing the current array makes sense.

**Comment by nlandolfi** [comment]
@viggyfresh please see new changes

**Pull request merged**